Personal finance expert system with group 9

OkekeKachi opened the pull request.

OkekeKachi committed:
Created Budget Planner with income/savings form, client-side validation, and dynamic pie chart generation for custom and recommended budgets. Includes basic styling and logo.

OkekeKachi committed:
made inputs clear and improved responsiveness

darlingxkill committed:
update to the budget rules funtion

darlingxkill committed:
Merge branch 'main' of https://github.com/OkekeKachi/Personal-Finance-Expert-System

darlingxkill committed:
added smart suggestions to the result page and improved budget logic

OkekeKachi committed:
added requirements

OkekeKachi committed:
Merge branch 'main' of https://github.com/OkekeKachi/Personal-Finance-Expert-System

imosudi commented:
Good afternoon,

I write to commend your effort and provide you with a quick review.

Your **`requiremnts.txt`** file was not properly generated, this was evident in the installation failure.

I modified the available **`requirements.txt`** to install the listed libraries, but the application failed to run.

darlingxkill committed:
Update README.md

OkekeKachi commented:
Good afternoon Sir,

Thank you for your feedback and for taking the time to review our work.

The requirements.txt file was initially generated manually, which resulted in some missing dependencies. I have now corrected this using the proper pip freeze method.

I also discovered that a conflicting push from one of my group members caused the application to break. I’ve resolved the conflict and applied a quick fix to ensure the application runs as expected.

Our current submission reflects the last commit made which was before the deadline. But, if I commit and push the updated fix, it'd be after the deadline. I apologize for the oversight and kindly request your permission to commit and push the corrected version

we are committed to delivering quality work and learning from this process.

darlingxkill committed:
Update README.md

OkekeKachi commented:
> Good afternoon,
> 
> I write to commend your effort and provide you with a quick review.
> 
> Your **`requiremnts.txt`** file was not properly generated, this was evident in the installation failure.
> 
> I modified the available **`requirements.txt`** to install the listed libraries, but the application failed to run.



Good afternoon Sir,

Thank you for your feedback and for taking the time to review our work.

The requirements.txt file was initially generated manually, which may have resulted in some missing dependencies. I have now corrected this using the proper pip freeze method.

I also discovered that a conflicting push from one of my group members was the main cause of the application to break. I’ve resolved the conflict and applied a quick fix to ensure the application runs as expected.

Our current submission reflects the last commit made which was before the deadline. But, if I commit and push the updated fix, it'd be after the deadline. I apologize for the oversight and kindly request your permission to commit and push the corrected version

we are committed to delivering quality work and learning from this process.

darlingxkill committed:
Update README.md

OkekeKachi committed:
updated requirements

imosudi commented:
Fix the following error and do a new PR:
![image](https://github.com/user-attachments/assets/f12ee62f-2472-49bf-a6f9-603cce8a6b37)

OkekeKachi commented:
Good day Sir,

Thank you once again for your feedback.

The error appears to be due to a Python version difference. Our project was built using Python 3.11.1, which still supports distutils. However, since you're using Python 3.12, where distutils has been removed, this likely caused the setup issue on your end.

You may choose to run the project using Python 3.11.1, which matches the original development environment and should eliminate the distutils error. Alternatively, we kindly request until the end of today to rebuild the project for compatibility with Python 3.12.

Apologies for the inconvenience, Sir.

imosudi commented:
I am perfectly OK. I could have corrected the errors on my own and moved on, but we need to do this to help you understand the subject matter better. There will be no need for any apology

Good job! Python 3.11 is OK. Your application came up smoothly. Ensure you fully notify people willing to run your application about such a limitation. There could be part of your dependencies that rely on or perform best on 3.11

Make use of the README.md file to create a good impression of your project, including the background studies of your work, the required libraries - dependencies, the programming language(s), and setup instructions

On the frontend, your form needs certain input types, especially numbers, but you should make your application a bit more flexible and friendly. You could use "text" input type and make use of JavaScript HTML form input type constraint enforcement. Try something like this and compare:
```
 <!DOCTYPE html>
 <html lang="en">
   <head>
       <meta charset="UTF-8">
       <meta name="viewport" content="width=device-width, initial-scale=1.0">
       <title>Numeric Input with onkeypress</title>
       <style>
           body {
               font-family: Arial, sans-serif;
               margin: 20px;
           }
           .error {
               color: red;
           }
       </style>
   </head>
   <body>
   
   <h2>Numeric Input Form</h2>
   <form id="numericForm">
       <label for="amount">Enter Amount (only figures):</label><br>
       <input type="text" id="amount" name="amount" required onkeypress="return isNumberKey(event)"><br>
       <span class="error" id="amountError"></span><br>
   
       <input type="submit" value="Submit">
   </form>
   
   <script>
       function isNumberKey(evt) {
           const charCode = (evt.which) ? evt.which : evt.keyCode;
           // Allow only numeric characters (0-9)
           if (charCode > 31 && (charCode < 48 || charCode > 57)) {
               return false; // Prevent the input
           }
           return true; // Allow the input
       }
   
       document.getElementById('numericForm').addEventListener('submit', function(event) {
           const amountInput = document.getElementById('amount');
           const amountError = document.getElementById('amountError');
           const amountValue = amountInput.value;
   
           // Clear previous error messages
           amountError.textContent = '';
   
           // Optionally validate on submit, if needed
           if (amountValue.trim() === '') {
               amountError.textContent = 'Amount cannot be empty.';
               event.preventDefault(); // Prevent form submission
           }
       });
   </script>
   </body>
 </html>

```
Meanwhile, the third group member is missing in action. What happened? Let me know if he/she prefers not to be graded.

OkekeKachi committed:
Enhance form validation and input flexibility

- Improved error handling for empty income, goal, and timeline fields.
- Updated input fields to use "text" type with JavaScript validation for numeric inputs.
- Added clear and user-friendly error messages for form submission.
- Updated README.md to include background, setup instructions, and dependency details.
- Notified users about Python 3.11 compatibility and possible limitations in the README.

OkekeKachi commented:
Thank you for your feedback, sir. I have implemented the necessary changes, including improving the form validation to prevent empty inputs and using JavaScript to enforce numeric input for better flexibility. I also updated the README.md with setup instructions, dependencies, and a note on Python 3.11 compatibility. I will ensure to notify users about these limitations going forward.

Also, that group member reached out to me about the project. Although he wasn’t able to contribute actively, I took time to explain our progress and current status to keep him informed.

Thank you for your guidance.

OkekeKachi committed:
update README
